src/atomate2/openmm/jobs/core.py

```diff
@@ -0,0 +1,250 @@
+from typing import List, Union, Optional, Dict, Tuple
+from pathlib import Path
+
+from dataclasses import dataclass
+
+from jobflow import Maker, job
+
+from pymatgen.io.openmm.generators import OpenMMSolutionGen
+from pymatgen.io.openmm.schema import InputMoleculeSpec
+from pymatgen.io.openmm.sets import OpenMMSet
+
+from openmm.openmm import (
+    MonteCarloBarostat,
+    AndersenThermostat,
+)
+from openmm.unit import (
+    kelvin,
+    atmosphere,
+    pico,
+    second
+)
+from openmm import Platform
+
+from src.atomate2.openmm.jobs.base import BaseOpenmmMaker
+
+import numpy as np
+
+
+@dataclass
+class InputMaker(Maker):
+    name: str = "input maker"
+
+    @job
+    def make(
+        self,
+        input_mol_dicts: List[Union[Dict, InputMoleculeSpec]],
+        density: Optional[float] = None,
+        box: Optional[List[float]] = None,
+        prev_dir: Optional[Union[str, Path]] = None
+    ):
+        """
+        Create a job for generating an OpenMMSet instance
+
+        Parameters
+        ----------
+        input_mol_dicts : List[Union[Dict, InputMoleculeSpec]]
+            List of dictionaries or InputMoleculeSpecs
+            for passing to OpenMMSolutionGen.
+        density : Optional[float]
+            Density of simulation, molcules/atoms per cubic centimeter.
+            Specify at most one argument between density and box.
+        box : Optional[List[float]
+            Dimensions of simulation box, in units of todo: ?.
+            Specify at most one argument between density and box.
+        prev_dir : Optional[Union[str, Path]]
+            Previous OpenMM simulation directory to link files from.
+        Returns
+        -------
+        Job
+            Job for generating an OpenMM input set instance.
+
+        """
+        input_set = OpenMMSolutionGen().get_input_set(
+            input_mol_dicts=input_mol_dicts,
+            density=density,
+            box=box,
+        )
+        return input_set
+
+
+@dataclass
+class EnergyMinimizationMaker(BaseOpenmmMaker):
+    name: str = "energy minimization"
+
+    @job
+    def make(self, input_set: OpenMMSet, prev_dir: Optional[Union[str, Path]] = None):
+        """
+        Create a flow that runs in the NPT ensemble.
+
+        Parameters
+        ----------
+        input_set : .OpenMMSet
+            A pymatgen structure object.
+        prev_dir : str or Path or None
+            A previous OpenMM simulation directory to link files from.
+
+        Returns
+        -------
+        Job
+            A OpenMM job containing one npt run.
+        """
+        platform = Platform.getPlatformByName("OpenCL")
```

Comment:
We should make platform selectable by the user here. OpenCL is just one possible platforms, could be CUDA or CPU too.

Reply:
This is handled in a future commit.